LazyProxy/LazyProxyBuilder.cs

```diff
@@ -64,9 +69,30 @@ public static Type GetType(Type type)
         /// <returns>The lazy proxy type instance.</returns>
         public static T CreateInstance<T>(Func<T> valueFactory)
         {
-            var lazy = new Lazy<T>(valueFactory);
-            var proxyType = GetType<T>();
-            return (T) Activator.CreateInstance(proxyType, lazy);
+            return (T) CreateInstance(typeof(T), () => valueFactory());
+        }
+
+        /// <summary>
+        /// Creates a lazy proxy type instance using a value factory.
+        /// </summary>
+        /// <param name="type">The interface proxy type implements.</param>
+        /// <param name="valueFactory">The function real value returns.</param>
+        /// <returns>The lazy proxy type instance.</returns>
+        public static object CreateInstance(Type type, Func<object> valueFactory)
+        {
+            var proxyType = GetType(type);
+
+            if (type.IsConstructedGenericType)
```

Comment:
I think it's better to move this logic to `GetType`, it does transition from closed-generic to open-generic, so why it shouldn't do reverse transition